source/agora/consensus/protocol/Nominator.d

```diff
@@ -367,22 +366,35 @@ extern(D):
     ***************************************************************************/
 
     public override void valueExternalized (uint64_t slot_idx,
-        ref const(Value) value)
+        ref const(Value) value) nothrow
     {
-        scope (failure) assert(0);
-
         if (slot_idx <= this.ledger.getBlockHeight())
             return;  // slot was already externalized
 
-        auto data = deserializeFull!ConsensusData(value[]);
+        ConsensusData data = void;
+        try
+            data = deserializeFull!ConsensusData(value[]);
+        catch (Exception exc)
+        {
+            log.fatal("Deserialization of C++ Value failed: {}", exc);
+            abort();
+        }
 
         // enrollment data may be empty, but not transaction set
         if (data.tx_set.length == 0)
             assert(0, "Transaction set empty");
 
         log.info("Externalized consensus data set at {}: {}", slot_idx, data);
-        if (!this.ledger.onExternalized(data))
-            assert(0);
+        try
+        {
+            if (!this.ledger.onExternalized(data))
+                assert(0);
```

Comment:
It seems like there should be an `abort` here too instead of the assert?

Reply:
Yeaaaahhhhh... This function now both throws and return a boolean. Will fix.